src/app/clusters/power-source-configuration-server/power-source-configuration-server.cpp

```diff
@@ -61,9 +64,27 @@ CHIP_ERROR PowerSourceConfigurationAttrAccess::Read(const ConcreteReadAttributeP
     {
     case Sources::Id:
         err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+            std::pair<uint16_t, uint8_t> orderEpPair[kMaxPowerSources];
+            uint8_t idx = 0, order = 0;
             for (auto endpoint : EnabledEndpointsWithServerCluster(PowerSource::Id))
             {
-                ReturnErrorOnFailure(encoder.Encode(endpoint));
+                if (idx >= kMaxPowerSources)
+                    break;
+                PowerSource::Attributes::Order::Get(endpoint, &order);
+                orderEpPair[idx] = std::make_pair(endpoint, order);
+                idx++;
+            }
+
+            int i, j, k;
+            for (i = 0; i < idx - 1; i++)
+            {
+                for (j = 0; j < idx - i - 1; j++)
+                    if ((orderEpPair[j]).second > (orderEpPair[j + 1]).second)
+                        swap(orderEpPair[j], orderEpPair[j + 1]);
+            }
+            for (k = 0; k < idx; k++)
+            {
+                ReturnErrorOnFailure(encoder.Encode(orderEpPair[k].first));
```

Comment:
```suggestion
            for (auto & pair : orderEpPair)
            {
                ReturnErrorOnFailure(encoder.Encode(pair.first));
```

Reply:
All in the list are not supposed to be encoded.